src/App/Fossa/FirstPartyScan.hs

```diff
@@ -0,0 +1,89 @@
+module App.Fossa.FirstPartyScan (
+  runFirstPartyScan,
+  firstPartyScanWithOrgInfo,
+) where
+
+import App.Fossa.Config.Analyze (StandardAnalyzeConfig (..), VendoredDependencyOptions (licenseScanPathFilters))
+import App.Fossa.LicenseScanner (scanVendoredDep)
+import App.Fossa.ManualDeps (VendoredDependency (..))
+import App.Types (FirstPartyScansFlag (..), FullFileUploads (FullFileUploads))
+import Control.Carrier.FossaApiClient (runFossaApiClient)
+import Control.Effect.Diagnostics (Diagnostics, fatalText)
+import Control.Effect.FossaApiClient (FossaApiClient, getOrganization)
+import Control.Effect.Lift (Lift)
+import Control.Effect.StickyLogger (StickyLogger)
+import Effect.Exec (Exec)
+import Effect.Logger (Logger, logDebug)
+import Effect.ReadFS (Has, ReadFS)
+import Fossa.API.Types (ApiOpts (..), Organization (..), blankOrganization)
+import Path (Abs, Dir, Path)
+import Srclib.Types (LicenseSourceUnit)
+
+runFirstPartyScan ::
+  ( Has Diagnostics sig m
+  , Has (Lift IO) sig m
+  , Has StickyLogger sig m
+  , Has Logger sig m
+  , Has Exec sig m
+  , Has ReadFS sig m
+  ) =>
+  Path Abs Dir ->
+  Maybe ApiOpts ->
+  StandardAnalyzeConfig ->
+  m (Maybe LicenseSourceUnit)
+runFirstPartyScan root maybeApiOpts cfg = do
+  -- if we do not have api opts, then we act as if the org defaults to not running first-party scans
+  -- but the FOSSA server supports first-party scans
+  case maybeApiOpts of
+    Nothing -> firstPartyScanMain root cfg defaultOrg
+    Just apiOpts -> runFossaApiClient apiOpts $ firstPartyScanWithOrgInfo root cfg
+  where
+    defaultOrg = blankOrganization{orgDefaultsToFirstPartyScans = False, orgSupportsFirstPartyScans = True, orgRequiresFullFileUploads = False}
```

Comment:
This is because we don't have an org when we run with the `--output` flag.

I've added that into the comment just above this. It looks like this now:

```haskell
  -- if we do not have api opts (i.e. the --output flag was used), then we act as if the org defaults to not running first-party scans
  -- but the FOSSA server supports first-party scans
```

Reply:
Ohhhh, I totally missed that the reason was for when we aren't uploading. Good comment 👍